lib/home/home_page_content.dart

```diff
@@ -39,6 +45,49 @@ class HomePageContent extends StatelessWidget {
             );
           }).toList(),
         );
+      } else if (section is ProductCarouselData) {
+        return BlocProvider<ProductListCubit>(
+          create: (_) => ProductListCubit()
+            ..loadProductsfromIds(productIds: section.productCodes),
+          child: BlocBuilder<ProductListCubit, ProductListState>(
+              builder: (context, state) {
+            return FlexCarousel(
+              floatingIndicator: false,
+              items: section.productCodes.map((code) {
+                final product =
+                    state.products.firstWhereOrNull((p) => p.code == code);
+
+                if (product != null) {
+                  return Column(
+                    children: [
+                      Expanded(
+                        child: CachedImage(
+                          url:
+                              '${dotenv.get('HYBRIS_BASE_URL')}${product.images.firstOrNull?.url}',
+                          fit: BoxFit.contain,
+                          placeholderAspectRatio: 1,
+                        ),
+                      ),
+                      Text(
+                        product.name,
+                        style: Theme.of(context).textTheme.bodyLarge,
+                      ),
+                      Text(product.price!.formattedValue),
+                    ],
+                  );
+                } else {
+                  return Container(
+                    height: 300,
+                    width: 400,
+                    child: const Center(
+                      child: CircularProgressIndicator(),
```

Comment:
Hmm.. if we can't find the product, CachedImage should handle that. Not sure we need this additional custom widget?

Reply:
The loader inside CachedImage is shown when the image is loading. This one is shown when the product is loading 🙂

Reply:
I admit that I could be nicer though, I can create a shimmer widget